### PR TITLE
testsuite-77: Fix race condition

### PR DESCRIPTION
**Pull request by daandemeyer**
Let's simplify the test and fix a race condition at the same time.

### CODE REVIEW

File: test/units/testsuite-77.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ read -r -u 4 text
 assert_eq "$text" "File"
 
 # Test for socket
-systemctl start testsuite-77-netcat.service
+echo "Socket" | nc -lkU /tmp/test.sock
```
Comment (bluca) [inline test/units/testsuite-77.sh]: won't this block?

Comment (daandemeyer) [reply test/units/testsuite-77.sh]: Yeah I just discovered that, any suggestions?

Comment (bluca) [reply test/units/testsuite-77.sh]: systemd-run ?

Comment (daandemeyer) [reply test/units/testsuite-77.sh]: Then you get the race condition again, ugh how hard can it be to write a message to a socket and continue lol

Comment (bluca) [reply test/units/testsuite-77.sh]: use the units again and order them?

Comment (daandemeyer) [reply test/units/testsuite-77.sh]: Still has a race condition without systemd-notify --ready, which we can't use because nc blocks

Comment (daandemeyer) [reply test/units/testsuite-77.sh]: I'll probably use a socket unit for this, will rework later

**Comment by DaanDeMeyer** [comment]
This version should do the trick, netcat is basically replaced with a inetd style socket+service so that we can start the socket separate and be sure that it actually exists before starting the client.

**Review by bluca — APPROVED** [review]

**Pull request merged**